include/deal.II/lac/petsc_compatibility.h

```diff
@@ -42,9 +51,116 @@ namespace PETScWrappers
                                 const std::string &value)
   {
 #if DEAL_II_PETSC_VERSION_LT(3, 7, 0)
-    PetscOptionsSetValue (name.c_str (), value.c_str ());
+    const PetscErrorCode ierr = PetscOptionsSetValue (name.c_str (), value.c_str ());
+#else
+    const PetscErrorCode ierr = PetscOptionsSetValue (NULL, name.c_str (), value.c_str ());
+#endif
+    (void)ierr;
+    Assert (ierr == 0, ExcPETScError(ierr));
+  }
+
+
+
+  /**
+   * Destroy a PETSc matrix. This function wraps MatDestroy with a version
+   * check (the signature of this function changed in PETSc 3.2.0).
+   *
+   * @warning Since the primary intent of this function is to enable RAII
+   * semantics in the PETSc wrappers, this function will not throw an
+   * exception if an error occurs, but instead just returns the error code
+   * given by MatDestroy.
+   *
+   */
+  inline PetscErrorCode destroy_matrix (Mat &matrix)
+  {
+    // PETSc will check whether or not matrix is NULL.
+#if DEAL_II_PETSC_VERSION_LT(3, 2, 0)
+    return MatDestroy (matrix);
+#else
+    return MatDestroy (&matrix);
+#endif
+  }
+
+
+
+  /**
+   * Destroy a Krylov Subspace (KSP) PETSc solver. This function wraps
+   * KSPDestroy with a version check (the signature of this function changed
+   * in PETSc 3.2.0).
+   *
+   * @warning Since the primary intent of this function is to enable RAII
+   * semantics in the PETSc wrappers, this function will not throw an
+   * exception if an error occurs, but instead just returns the error code
+   * given by MatDestroy.
+   */
+  inline PetscErrorCode destroy_krylov_solver (KSP &krylov_solver)
+  {
+    // PETSc will check whether or not matrix is NULL.
+#if DEAL_II_PETSC_VERSION_LT(3, 2, 0)
+    return KSPDestroy (krylov_solver);
+#else
+    return KSPDestroy (&krylov_solver);
+#endif
+  }
+
+
+
+  /**
+   * Set a PETSc matrix option. This function wraps MatSetOption with a
+   * version check.
+   *
+   * @warning The argument option_value is ignored in versions of PETSc
+   * before 3.0.0 since the corresponding function did not take this argument.
+   */
+  inline void set_matrix_option (Mat &matrix,
+                                 const MatOption option_name,
+                                 const PetscBooleanType option_value = PETSC_FALSE)
+  {
+#if DEAL_II_PETSC_VERSION_LT(3,0,0)
+    const int ierr = MatSetOption (matrix, option_name);
```

Comment:
same here. We could make the code so much simpler if we got rid of backward compatibility with really old versions of PETSc.